### ISSUE TITLE
Mean and variance at the training points

### ISSUE THREAD

**Issue by rcnlee** [STOR-i/GaussianProcesses.jl#84]
Is there an easy way to get the mean and variance at the training points?  Is it already available in ``gp`` after calling ``fit!`` (thus saving compute)? or do I need to use ``predict_y`` to compute it?

Thanks!

**Comment by maximerischard** [comment]
No, this isn't implemented, and I agree that it should, as recomputing the covariance matrix is wasteful.

**Comment by path-doc** [comment]
I posted this question on Julia discourse:

>As part of a broader Value Function Iteration exercise, I wish to feed the output of `predictMVN` or `predict_f` back into an instantiation of `GP` .
>Eg. Suppose we have created an instance `gp` by feeding `GP` the usual ingredients: training data `x1` , target data `t1` , along with eg `MeanPoly(B)` and a suitable kernel `kern` .
>After optimising or fitting, how would I extract the posterior mean and covariance functions from `gp` and, moreover, extract them in as type `Mean` and `Kernel` respectively?

@maximerischard, from what you're saying, I should currently be able to recompute. Can you explain how one does that? I might then attempt to fix this issue.
